tedana/decomposition/ma_pca.py

```diff
@@ -0,0 +1,600 @@
+                         'independent samples.')
+    n_iters = n_iters_0
+    LGR.debug('Estimated the entropy rate of the Gaussian component '
+              'with subsampling depth {}'.format(j + 1))
+
+    return n_iters, ent_rate
+
+
+def _subsampling(data, sub_depth):
+    """
+    Subsampling the data evenly with space 's'.
+
+    Parameters
+    ----------
+    data : ndarray
+        The data to be subsampled
+    sub_depth : int
+        The subsampling depth
+
+    Returns
+    -------
+    out : ndarray
+        Subsampled data
+    """
+
+    # First index from which to start subsampling for each dimension
+    idx_0 = [0, 0, 0]
+    ndims = data.shape
+
+    if data.ndim == 3 and np.min(ndims) != 1:  # 3D
+        out = data[np.arange(
+            idx_0[0], ndims[0], sub_depth), :, :][:, np.arange(
+                idx_0[1], ndims[1], sub_depth), :][:, :, np.arange(idx_0[2], ndims[2], sub_depth)]
+    else:
+        raise ValueError('Unrecognized matrix dimension! )'
+                         'Input array must be 3D with min dimension > 1.')
+
+    return out
+
+
+def _kurtn(data):
+    """
+    Normalized kurtosis funtion so that for a Gaussian r.v. the kurtn(g) = 0.
+
+    Parameters
+    ----------
+    data : ndarray
+        The data to calculate the kurtosis of
+
+    Returns
+    -------
+    kurt : (1:N) array-like
+        The kurtosis of each vector in x along the second dimension. For
+        tedana, this will be the kurtosis of each PCA component.
+    """
+
+    kurt = np.zeros((data.shape[1], 1))
+
+    for i in range(data.shape[1]):
+        data_norm = detrend(data[:, i], type='constant')
+        data_norm /= np.std(data_norm)
+        kurt[i] = np.mean(data_norm**4) - 3
+
+    kurt[kurt < 0] = 0
+
+    return kurt
+
+
+def _icatb_svd(data, n_comps=None):
+    """
+    Run Singular Value Decomposition (SVD) on input data and extracts the
+    given number of components (n_comps).
+
+    Parameters
+    ----------
+    data : array
+        The data to compute SVD for
+    n_comps : int
+        Number of PCA components to be kept
+
+    Returns
+    -------
+    V : 2D array
+        Eigenvectors from SVD
+    Lambda : float
+        Eigenvalues
+    """
+
+    if not n_comps:
+        n_comps = np.min(data.shape[0], data.shape[1])
+
+    _, Lambda, vh = svd(data, full_matrices=False)
+
+    # Sort eigen vectors in Ascending order
+    V = vh.T
+    Lambda = Lambda / np.sqrt(data.shape[0] - 1)  # Whitening (sklearn)
+    inds = np.argsort(np.power(Lambda, 2))
+    Lambda = np.power(Lambda, 2)[inds]
+    V = V[:, inds]
+    sumAll = np.sum(Lambda)
+
+    # Return only the extracted components
+    V = V[:, (V.shape[1] - n_comps):]
+    Lambda = Lambda[Lambda.shape[0] - n_comps:]
+    sumUsed = np.sum(Lambda)
+    retained = (sumUsed / sumAll) * 100
+    LGR.debug('{ret}% of non-zero components retained'.format(ret=retained))
+
+    return V, Lambda
+
+
+def _eigensp_adj(lam, n, p):
```

Comment:
Here, could you actually put the Note as the top line of the docstring and put that reference in a References section instead of notes?